File: newproject2/app/src/main/java/com/example/newproject2/MainActivity.java
package com.example.newproject2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.webkit.WebView;
import android.webkit.WebViewClient;

public class MainActivity extends AppCompatActivity {

    WebView wv;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        wv=(WebView)findViewById(R.id.web1);
        wv.setWebViewClient(new WebViewClient());// alots a space to the screen

        wv.loadUrl("https://www.facebook.com/oncountrywork/");// loads the url to the screen
    }
}

File: API_Retrofit/app/src/main/java/com/example/api_retrofit/MainActivity.kt
package com.example.api_retrofit

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.coroutines.flow.callbackFlow
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        getData()

    }

    fun getData() {



RetroFitInstance.apiInterfce.getData().enqueue(object : Callback<responseDataClass> {
    override fun onResponse(call: Call<responseDataClass>, response: Response<responseDataClass>) {
        if (response.isSuccessful) {
            val data = response.body() // Handle the response data here
            // Do something with the data
        } else {
            // Handle unsuccessful response
        }
    }

    override fun onFailure(call: Call<responseDataClass>, t: Throwable) {
        // Handle failure/error here
    }
})




    }

}






